code/items/weapons/generics/TTTWeaponBase.cs

```diff
@@ -0,0 +1,83 @@
+using System;
```

Comment:
Is this class / file used?

Reply:
Yeah, everything inside of `generics` is being used. So `SWB` has multiple derived classes called `SWB_Base.WeaponBase`, `SWB_Base.WeaponBaseEntity`, `SWB_Base.WeaponBaseMelee`, etc.

I needed to make a class to derive from each of these classes so we could put our `TTT` data on them (ex. `TTTWeaponBase`, `TTTWeaponBaseMelee`, etc.). 

The other option would be to do `composition`, where a new `TTTWeaponData` class is made and then placed inside of each one of these classes. This might be a bit cleaner, but the `TTTWeaponBaseGeneric` cleans it up a bit.